src/auth/state/app.ts

```diff
@@ -0,0 +1,61 @@
+import FileSystem from "../../fs/index.js"
+
+
+// SCENARIO
+
+
+export enum AppScenario {
+  NotAuthed = "NOT_AUTHORIZED",
+  Authed = "AUTHORIZED"
+}
+
+export function scenarioAuthed(
+  username: string,
+  fs: FileSystem | undefined
+): Authed {
+  return {
+    kind: "appState",
+    scenario: AppScenario.Authed,
+
+    authenticated: true,
+    fs,
+    username
+  }
+}
+
+export function scenarioNotAuthed(): NotAuthed {
+  return {
+    kind: "appState",
+    scenario: AppScenario.NotAuthed,
+
+    authenticated: false
+  }
+}
+
+
+
+// STATE
+
+
+export type AppState
+  = NotAuthed
+  | Authed
```

Comment:
The only inconsistency here is that "continuation" is missing (ie. was authed before, continuing existing sessions). Was that intentional?

Reply:
Yep, intentional. `Authed` is roughly `Continuation`.

We don't need an `AuthSucceeded` because the user will register without a page load and re-initialization of webnative. `Continuation` can be renamed to `Authed` because we don't need to differentiate it from `AuthSucceeded`.